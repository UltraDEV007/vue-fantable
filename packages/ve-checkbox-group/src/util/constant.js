// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
