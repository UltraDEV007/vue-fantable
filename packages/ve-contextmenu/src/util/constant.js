// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
