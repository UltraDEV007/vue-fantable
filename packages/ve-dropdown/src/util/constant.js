// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
