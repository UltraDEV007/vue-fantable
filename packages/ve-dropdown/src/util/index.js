/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
