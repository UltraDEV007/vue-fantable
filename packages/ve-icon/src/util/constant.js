// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
