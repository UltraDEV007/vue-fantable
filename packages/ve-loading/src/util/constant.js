// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
