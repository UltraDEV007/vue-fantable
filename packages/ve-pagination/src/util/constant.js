// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
