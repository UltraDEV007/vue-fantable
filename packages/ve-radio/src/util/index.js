import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
