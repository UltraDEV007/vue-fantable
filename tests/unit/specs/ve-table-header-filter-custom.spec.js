import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import veIcon from "@/ve-table";
import { later } from "../util";

describe("veTable header filter custom", () => {
    const mockFilterFn = jest.fn((closeFn) => closeFn());
    const mockBeforeVisibleChangeFn = jest.fn();

    const TABLE_DATA = [
        {
            name: "John",
            date: "1900-05-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shanghai",
            rowkey: 0,
        },
        {
            name: "Dickerson",
            date: "1910-06-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Beijing",
            rowkey: 1,
        },
        {
            name: "Larsen",
            date: "2000-07-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Chongqing",
            rowkey: 2,
        },
        {
            name: "Geneva",
            date: "2010-08-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Xiamen",
            rowkey: 3,
        },
        {
            name: "Jami",
            date: "2020-09-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
            rowkey: 4,
        },
    ];

    afterEach(() => {
        return new Promise((resolve) => {
            const poppers = document.querySelector(".ve-dropdown-popper");
            poppers && poppers.remove();

            resolve();
        });
    });

    it("render", () => {
        const wrapper = mount({
            render() {
                return (
                    <veTable
                        columns={this.columns}
                        tableData={this.tableData}
                    />
                );
            },
            data() {
                return {
                    searchValue: "",
                    columns: [
                        {
                            field: "name",
                            key: "a",
                            title: "Name",
                            align: "left",
                            width: "15%",
                            // filter custom
                            filterCustom: {
                                defaultVisible: true,
                                render: ({ showFn, closeFn }, h) => {
                                    return (
                                        <div class="custom-name-filter">
                                            <input
                                                value={this.searchValue}
                                                on-input={(e) =>
                                                    (this.searchValue =
                                                        e.target.value)
                                                }
                                                placeholder="Search name"
                                            />
                                            <div class="custom-name-filter-operation">
                                                <span
                                                    class="name-filter-cancel"
                                                    on-click={() =>
                                                        mockFilterFn(closeFn)
                                                    }
                                                >
                                                    取消
                                                </span>
                                                <span
                                                    class="name-filter-confirm"
                                                    on-click={() =>
                                                        mockFilterFn(closeFn)
                                                    }
                                                >
                                                    查询
                                                </span>
                                            </div>
                                        </div>
                                    );
                                },
                                // custom filter icon
                                filterIcon: (h) => {
                                    return <ve-icon name="search" />;
                                },
                            },
                        },
                        {
                            field: "date",
                            key: "b",
                            title: "Date",
                            align: "left",
                            width: "15%",
                        },
                        {
                            field: "hobby",
                            key: "c",
                            title: "Hobby",
                            align: "center",
                            width: "30%",
                        },
                        {
                            field: "address",
                            key: "d",
                            title: "Address",
                            align: "left",
                            width: "40%",
                        },
                    ],
                    tableData: TABLE_DATA,
                };
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("custom filter", async () => {
        const wrapper = mount(
            {
                render() {
                    return (
                        <veTable
                            columns={this.columns}
                            tableData={this.tableData}
                        />
                    );
                },
                data() {
                    return {
                        searchValue: "",
                        columns: [
                            {
                                field: "name",
                                key: "a",
                                title: "Name",
                                align: "left",
                                width: "15%",
                                // filter custom
                                filterCustom: {
                                    defaultVisible: false,
                                    render: ({ showFn, closeFn }, h) => {
                                        return (
                                            <div class="custom-name-filter">
                                                <input
                                                    value={this.searchValue}
                                                    on-input={(e) =>
                                                        (this.searchValue =
                                                            e.target.value)
                                                    }
                                                    placeholder="Search name"
                                                />
                                                <div class="custom-name-filter-operation">
                                                    <span
                                                        class="name-filter-cancel"
                                                        on-click={() =>
                                                            mockFilterFn(
                                                                closeFn,
                                                            )
                                                        }
                                                    >
                                                        取消
                                                    </span>
                                                    <span
                                                        class="name-filter-confirm"
                                                        on-click={() =>
                                                            mockFilterFn(
                                                                closeFn,
                                                            )
                                                        }
                                                    >
                                                        查询
                                                    </span>
                                                </div>
                                            </div>
                                        );
                                    },
                                },
                            },
                            {
                                field: "date",
                                key: "b",
                                title: "Date",
                                align: "left",
                                width: "15%",
                            },
                            {
                                field: "hobby",
                                key: "c",
                                title: "Hobby",
                                align: "center",
                                width: "30%",
                            },
                            {
                                field: "address",
                                key: "d",
                                title: "Address",
                                align: "left",
                                width: "40%",
                            },
                        ],
                        tableData: TABLE_DATA,
                    };
                },
            },
            {
                attachTo: document.body,
            },
        );

        // default icon
        expect(wrapper.find(".icon-vet-filter").exists()).toBe(true);

        wrapper.find(".ve-table-filter-icon").trigger("click");
        // 延迟展示
        await later(100);

        const showDropdownPopper = document.querySelector(
            ".ve-dropdown-popper > .ve-dropdown-dd",
        );

        expect(showDropdownPopper.textContent.length).toBeGreaterThan(1);

        const cancelClickEvent = new MouseEvent("click", {
            view: window, // window
            bubbles: true,
            cancelable: true,
        });

        document
            .querySelector(".name-filter-cancel")
            .dispatchEvent(cancelClickEvent);

        expect(mockFilterFn).toBeCalled();
        expect(mockFilterFn).toHaveBeenCalledWith(expect.any(Function));
    });

    it("beforeVisibleChange callback method", async () => {
        const wrapper = mount(
            {
                render() {
                    return (
                        <veTable
                            columns={this.columns}
                            tableData={this.tableData}
                        />
                    );
                },
                data() {
                    return {
                        searchValue: "",
                        columns: [
                            {
                                field: "name",
                                key: "a",
                                title: "Name",
                                align: "left",
                                width: "15%",
                                // filter custom
                                filterCustom: {
                                    defaultVisible: false,
                                    beforeVisibleChange: ({ nextVisible }) => {
                                        mockBeforeVisibleChangeFn({
                                            nextVisible,
                                        });
                                    },
                                    render: ({ showFn, closeFn }, h) => {
                                        return (
                                            <div class="custom-name-filter">
                                                <input
                                                    value={this.searchValue}
                                                    on-input={(e) =>
                                                        (this.searchValue =
                                                            e.target.value)
                                                    }
                                                    placeholder="Search name"
                                                />
                                                <div class="custom-name-filter-operation">
                                                    <span
                                                        class="name-filter-cancel"
                                                        on-click={() =>
                                                            mockFilterFn(
                                                                closeFn,
                                                            )
                                                        }
                                                    >
                                                        取消
                                                    </span>
                                                    <span
                                                        class="name-filter-confirm"
                                                        on-click={() =>
                                                            mockFilterFn(
                                                                closeFn,
                                                            )
                                                        }
                                                    >
                                                        查询
                                                    </span>
                                                </div>
                                            </div>
                                        );
                                    },
                                },
                            },
                            {
                                field: "date",
                                key: "b",
                                title: "Date",
                                align: "left",
                                width: "15%",
                            },
                            {
                                field: "hobby",
                                key: "c",
                                title: "Hobby",
                                align: "center",
                                width: "30%",
                            },
                            {
                                field: "address",
                                key: "d",
                                title: "Address",
                                align: "left",
                                width: "40%",
                            },
                        ],
                        tableData: TABLE_DATA,
                    };
                },
            },
            {
                attachTo: document.body,
            },
        );

        // default icon
        expect(wrapper.find(".icon-vet-filter").exists()).toBe(true);

        wrapper.find(".ve-table-filter-icon").trigger("click");
        // 延迟展示
        await later(100);

        expect(mockBeforeVisibleChangeFn).toBeCalled();
        expect(mockBeforeVisibleChangeFn).toHaveBeenCalledWith({
            nextVisible: true,
        });

        const showDropdownPopper = document.querySelector(
            ".ve-dropdown-popper > .ve-dropdown-dd",
        );

        expect(showDropdownPopper.textContent.length).toBeGreaterThan(1);

        const cancelClickEvent = new MouseEvent("click", {
            view: window, // window
            bubbles: true,
            cancelable: true,
        });

        await later();

        expect(mockBeforeVisibleChangeFn).toBeCalled();
        expect(mockBeforeVisibleChangeFn).toHaveBeenCalledWith({
            nextVisible: false,
        });
    });
});
