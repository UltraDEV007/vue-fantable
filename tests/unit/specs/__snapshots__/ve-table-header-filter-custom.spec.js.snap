// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veTable header filter custom render 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 15%;">
            <col style="width: 15%;">
            <col style="width: 30%;">
            <col style="width: 40%;">
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name<dl class="ve-dropdown">
                  <dt class="ve-dropdown-dt"><a class="" style="width: 90px;"><span class="fan-table-filter"><span class="fan-table-filter-icon"><i class="ve-icon iconfont-vet icon-vet-search"></i></span></span></a></dt>
                  <div style="display: none;">
                    <dd class="ve-dropdown-dd ve-dropdown-dd-show">
                      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
                        <div>
                          <div class="custom-name-filter"><input placeholder="Search name">
                            <div class="custom-name-filter-operation"><span class="name-filter-cancel">取消</span><span class="name-filter-confirm">查询</span></div>
                          </div>
                        </div>
                      </ul>
                    </dd>
                  </div>
                </dl>
              </th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Date</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: left; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">John</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1900-05-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shanghai</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1910-06-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Larsen</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2000-07-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Chongqing</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Geneva</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2010-08-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Xiamen</td>
            </tr>
            <tr class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Jami</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2020-09-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: left;">No.1 Century Avenue, Shenzhen</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
      </div>
    </div>
  </div>
</div>
`;
